### ISSUE TITLE
Option to Toggle Conjugation/Declension Autogen Overwrite of Word while Generating

### ISSUE THREAD

**Issue by bobby-bolton** [DraqueT/PolyGlot#948]
TL;DR: When autogenerating conjugations, let there be a toggle to make the word created in each step of the autogen be the new word for comparing rules.

The title isn't the best, and I know this isn't the best way of expressing the idea, but it's the best I've got. This is what I mean:

My current language, currently unnamed, forms genitives the following way (i.e. the order of rules in the conj/decl autogen):
- First, it adds the "accusative suffix", which is used for many cases. There's no problem here.
- Next, it adds the "genitive prefix", which adds _mi_ if the word begins with a consonant and just _m_ if it begins with a vowel. The problem lies in the next step...
- Lastly, it adds the article. This step itself is rather complex, and is where the problem occurs.
  - If the word begins with a vowel, add either _il-_ for definite or _ul-_ for indefinite.
  - If the word begins with a consonant, add _iX-_ for definite or _uX-_ for indefinite. _X_ represents the first letter of the word itself. This has no problems by itself.

The problem lies in the fact that the last rule only checks the base form of the word and not the newest form, i.e. the one created in the autogen. A word like "aba" would go:
- Accusative suffix for **aba** is _m_, so the newest word is "abam"
- Genitive prefix for **aba** is _m-_, so the newest word is "mabam"
- Article for **aba** is _il-_... except that the newest word, "mabam", no longer matches the vowel rule of **aba**, and **aba** no longer matches the consonant rule... so both fail.

I understand that this fact is utilized by at least one language in Polyglot, which is why I ask there to be a toggle for making the newest word be the word that the rules compare with. This would mean that **aba** would be conjugated (properly) as _im-mabam_.

I understand that this may take a while, but please try to do this as quickly as you can, since I do not wish to leave my language hanging. Anyways, thank you for your hard work in building this program; it really is a (paper) life saver!

**Comment by DraqueT** [comment]
This is a really good idea. I will definitely include it in the next major release of PolyGlot.

**Comment by bobby-bolton** [comment]
> This is a really good idea. I will definitely include it in the next major release of PolyGlot.

That's really nice to hear! Thank you!
